src/physics/systems.rs

```diff
@@ -60,74 +63,119 @@ pub fn create_joints_system(
 
 /// System responsible for performing one timestep of the physics world.
 pub fn step_world_system(
+    (time, mut sim_to_render_time): (Res<Time>, ResMut<SimulationToRenderTime>),
     configuration: Res<RapierConfiguration>,
     integration_parameters: Res<IntegrationParameters>,
-    mut pipeline: ResMut<PhysicsPipeline>,
-    mut query_pipeline: ResMut<QueryPipeline>,
-    mut broad_phase: ResMut<BroadPhase>,
-    mut narrow_phase: ResMut<NarrowPhase>,
+    (mut pipeline, mut query_pipeline): (ResMut<PhysicsPipeline>, ResMut<QueryPipeline>),
+    (mut broad_phase, mut narrow_phase): (ResMut<BroadPhase>, ResMut<NarrowPhase>),
     mut bodies: ResMut<RigidBodySet>,
     mut colliders: ResMut<ColliderSet>,
     mut joints: ResMut<JointSet>,
     events: Res<EventQueue>,
+    mut query: Query<(
+        &RigidBodyHandleComponent,
+        &mut PhysicsInterpolationComponent,
+    )>,
 ) {
     if events.auto_clear {
         events.clear();
     }
 
-    if configuration.physics_pipeline_active {
-        pipeline.step(
-            &configuration.gravity,
-            &integration_parameters,
-            &mut broad_phase,
-            &mut narrow_phase,
-            &mut bodies,
-            &mut colliders,
-            &mut joints,
-            &*events,
-        );
+    sim_to_render_time.diff += time.delta_seconds;
+
+    let sim_dt = integration_parameters.dt();
+    while sim_to_render_time.diff > sim_dt {
+        if configuration.physics_pipeline_active {
+            if sim_to_render_time.diff < 2.0 * sim_dt {
```

Comment:
I think it is best to do the comparison that way:

```suggestion
            // NOTE: in this comparison we do the same computations we
            // will do for the next `while` iteration test, to make sure we
            // don't get bit by potential float inaccuracy.
            if sim_to_render_time.diff - simd_dt < sim_dt {
```

This ensures we perform the comparison the same way as for the `while` statement (with the same float operations). That way there is no risk of floating-point errors making this case succeed/fail when we would not expect it to.

Reply:
There was a typo in your suggestion so I added it manually. Thanks!